_posts/news/2025-04-21-announcing-scalajs-1.19.0.md

```diff
@@ -0,0 +1,181 @@
+If that setting is used through an indirect dependency, it will be silently ignored.
+
+Strict floats are almost entirely backward compatible with non-strict floats.
+The latter were mostly specified on a best-effort basis, which strict floats comply with.
+The only exception is that tests of the form `x.isInstanceOf[Float]` (or `case x: Float =>`) will answer `false` for number values that cannot exactly be represented in a 32-bit `Float`.
+
+We are not aware of any codebase that was ever adversely affected by strict float semantics.
+
+### Changes to the IR and linker APIs
+
+For tooling authors who directly manipulate the IR and linker APIs, there have been some breaking changes in that area.
+This is in line with our version policy for the linker APIs.
+
+The most likely changes you may hit are:
+
+* The `ir.Trees.Closure` node has two additional fields: `flags` and `resultType`.
+* All the "well-known" names that were previously in `ir.Names` have been moved to a new object `ir.WellKnownNames`.
+* `ir.Types.PrimRef` is not a case class anymore.
+
+## Enhancements
+
+### Native support for JS `async/await`
+
+Note 1: Scala 3 users will have to wait for a future release of Scala 3---likely 3.8.0---to be able to use this new feature.
+
+Note 2: when targeting WebAssembly, this feature requires support for JSPI (JavaScript Promise Integration).
+You can [check support for JSPI in various browsers here](https://webassembly.org/features/#table-row-jspi).
+On Node.js v23+, this requires the flag `--experimental-wasm-jspi`.
+
+This release adds a pair of primitives, `js.async` and `js.await`, which correspond to the `async/await` feature of JavaScript.
+In order to use these functions, you must configure Scala.js to target ECMAScript 2017 or later.
+In an sbt build, use the following setting:
+
+{% highlight scala %}
+// Target ES 2017 to enable async/await
+scalaJSLinkerConfig := {
+  scalaJSLinkerConfig.value
+    .withESFeatures(_.withESVersion(ESVersion.ES2017)) // enable async/await
+},
+{% endhighlight %}
+
+You can then use `js.async { ... }` blocks containing `js.await` calls.
+For example:
+
+{% highlight scala %}
+val p: js.Promise[String] = downloadSomething()
+val result: js.Promise[Int] = js.async {
+  val text: String = js.await(p)
+  text.toInt
+}
+{% endhighlight %}
+
+`js.async { ... }` executes a block of code under a JavaScript `async` context.
+The block of code can await `js.Promise`s using `js.await`.
+Doing so will continue after the call to `js.await` when the given `Promise` is resolved.
+If the `Promise` is rejected, the exception gets rethrown at the call site.
+
+A block such as `js.async { body }` is equivalent to an immediately-applied JavaScript `async` function:
+
+{% highlight javascript %}
+(async () => body)()
+{% endhighlight %}
+
+`js.async` returns a `js.Promise` that will be resolved with the result of the code block.
+If the block throws an exception, the `Promise` will be rejected.
```

Comment:
Are you sure this is true for exceptions thrown before the first await?

Reply:
Yes. Both my understanding of the spec and experimentation agree: the `Promise` is rejected if there is an exception before the first `await`. Both in JS and in Wasm/JSPI.